source/MapPanel.cpp

```diff
@@ -1045,7 +1045,7 @@ void MapPanel::DrawEscorts()
 void MapPanel::DrawWormholes()
 {
 	// Keep track of what arrows and links need to be drawn.
-	set<pair<const System *, const System *>> arrowsToDraw;
+	set<pair<pair<const System *, const System *>, const Color *>> arrowsToDraw;
```

Comment:
so would I a pair is already pretty bad but so long as it is just number/thing or twice the same thing ok but if it gets worse than that...
can't we somehow use the wormhole class for this?

Reply:
No we can't use the wormhole class, because that's overkill. Just make it a struct local to this file, maybe something like this:

```c++
struct WormholeArrow {
    const System *from = nullptr;
    const System *to = nullptr;
    const Color *color = nullptr;
};
```

Reply:
```c++
struct WormholeArrow {
    WormholeArrow(const System *from, const System *to, const Color *color)
    : from(from), to(to), color(color) {}
    const System *from = nullptr;
    const System *to = nullptr;
    const Color *color = nullptr;
};
```

give it a constructor, for easier inserting in a set